scipy/optimize/_sda.py

```diff
@@ -0,0 +1,572 @@
+# Simulated Dual Annealing implementation.
+# Copyright (c) 2016 Sylvain Gubian <sylvain.gubian@pmi.com>,
+# Yang Xiang <yang.xiang@pmi.com>
+# Author: Sylvain Gubian, PMP S.A.
+"""
+SDA: A Simulated Dual Annealing global optimization algorithm
+"""
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from scipy.optimize import OptimizeResult
+from scipy.optimize import minimize
+from scipy.optimize._numdiff import approx_derivative
+from scipy.special import gammaln
+from scipy._lib._util import check_random_state
+
+__all__ = ['sda']
+BIG_VALUE = 1e16
+
+
+class VisitingDistribution(object):
+    """
+    Class used to generate new coordinates based on the distorted
+    Cauchy-Lorentz distribution. Depending on the steps within the Markov
+    chain, the class implements the strategy for generating new location
+    changes.
+    """
+    tail_limit = 1.e8
+    min_visit_bound = 1.e-10
+
+    def __init__(self, lb, ub, qv, rs):
+        self.qv = qv
```

Comment:
I would strongly suggest that you use descriptive variable names.  I have no clue just looking at this what qv, rs, b_range, x_gauss, s_gauss are.  Same goes for SDARunner.es and many of the other SDARunner variables

Reply:
Ok. I will rename them (except for qv, this is the equation term name, maybe, cannot really do better)